flocker/control/test/test_model.py

```diff
@@ -1460,3 +1462,190 @@ def test_case_insensitive(self):
         link = Link(alias=u'myLINK', local_port=1, remote_port=1)
         link2 = link.set('alias', u'MYlink')
         self.assertEqual(link, link2)
+
+
+class LeaseTests(SynchronousTestCase):
+    """
+    Tests for ``Leases``.
+    """
+    def setUp(self):
+        """
+        Setup for each test.
+        """
+        self.leases = Leases()
+        self.now = datetime.datetime.now()
+        self.dataset_id = uuid4()
+        self.node_id = uuid4()
+        self.dataset = Dataset(dataset_id=unicode(self.dataset_id))
+        self.node = Node(uuid=self.node_id)
+        self.lease_duration = 60 * 60
+
+    def test_lease_expiry_datetime(self):
+        """
+        An lease has an expiry date/time after the specified number
+        of seconds from the time of acquisition.
+        """
+        expected_expiration = self.now + datetime.timedelta(
+            seconds=self.lease_duration)
+        leases = self.leases.acquire(
+            self.now, self.dataset_id, self.node_id, self.lease_duration
+        )
+        lease = leases.get(self.dataset_id)
+        self.assertEqual(lease.expiration, expected_expiration)
+
+    def test_indefinite_lease(self):
+        """
+        An acquired lease can be set to never expire.
+        """
+        leases = self.leases.acquire(self.now, self.dataset_id, self.node_id)
+        lease = leases.get(self.dataset_id)
+        self.assertIsNone(lease.expiration)
+
+    def test_lease_expires(self):
+        """
+        An acquired lease expires after the specified number of seconds and
+        is removed from the ``Leases`` map.
+        """
+        leases = self.leases.acquire(
+            self.now, self.dataset_id, self.node_id, self.lease_duration
+        )
+        # Assert the lease has been acquired successfully.
+        self.assertTrue(self.dataset_id in leases)
```

Comment:
`assertIn(self.dataset_id, leases)` will get you a better error message when this fails.